spec/justifi/payment_method_spec.rb

```diff
@@ -16,13 +16,61 @@
     let(:justifi_object) { list_payment_methods }
 
     context "with valid params" do
+      before do
+        Stubs::PaymentMethod.empty_list
+      end
+
+      it do
+        expect(justifi_object).to be_a(Justifi::ListObject)
+        expect(justifi_object.empty?).to be_truthy
+      end
+    end
+
+    context "with next_page" do
       before do
         Stubs::PaymentMethod.success_list
       end
 
+      let(:object) { justifi_object }
+      let(:next_card_id) { "pm_1234xyz" }
+      let(:next_page) { object.next_page }
+
       it do
-        expect(justifi_object).to be_a(Justifi::JustifiObject)
-        expect(justifi_object.raw_response.http_status).to eq(200)
+        Stubs::PaymentMethod.success_list({after_cursor: object.end_cursor}, next_card_id)
+        object.each do |payment_method|
+          expect(payment_method[:card][:id]).not_to eq(next_card_id)
+        end
+        expect(next_page.data.first[:card][:id]).to eq(next_card_id)
+      end
+    end
+
+    context "with previous_page" do
+      before do
+        Stubs::PaymentMethod.success_list
+      end
+
+      let(:object) { justifi_object }
+      let(:card_id) { "pm_1234xyz" }
+      let(:previous_page) { object.previous_page }
+
+      it do
+        Stubs::PaymentMethod.success_list({before_cursor: object.start_cursor}, card_id)
+        expect(object.data.first[:card][:id]).not_to eq(previous_page.data.first[:card][:id])
+      end
+    end
+
+    context "when next_page is empty" do
+      before do
+        Stubs::PaymentMethod.success_list
+        allow_any_instance_of(Justifi::ListObject).to receive(:has_next).and_return(false)
+      end
+
+      let(:previous_data) { justifi_object }
```

Comment:
How about renaming this to `:current_data` instead? I was confused for a moment after reading the test for previous page above .

Reply:
Good idea!